pkg/config/mesh/mesh_test.go

```diff
@@ -89,14 +96,22 @@ func TestApplyProxyConfig(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
-		// Ensure we didn't modify the passed in mesh config
-		if !reflect.DeepEqual(mc.DefaultConfig.ProxyMetadata, map[string]string{
-
+		// Ensure we did modify the mesh config
+		expected := map[string]string{
 			"merged":   "override",
 			"default":  "foo",
 			"override": "bar",
-		}) {
-			t.Fatalf("unexpected proxy metadata: %+v", mc.DefaultConfig.ProxyMetadata)
+		}
+		if !reflect.DeepEqual(mc.DefaultConfig.ProxyMetadata, expected) {
+			t.Fatalf("unexpected proxy metadata. expected: %+v, actual: %+v", expected, config.DefaultConfig.ProxyMetadata)
+		}
+		// Ensure we didn't modify the default mesh config
+		expected = map[string]string{
+			"merged":  "original",
+			"default": "foo",
+		}
```

Comment:
Sorry I missed this. This test is calling `ApplyMeshConfig` with changed config - So it will override. Which is correct. I guess no changes needed? This expectation is incorrect.

Reply:
this test calls `ApplyProxyConfig` which the comment above it says 
`// ApplyProxyConfig applies the give proxy config yaml to a mesh config object. The passed in mesh config
// will not be modified.`

Reply:
Is n't this calling  ApplyMeshConfig https://github.com/istio/istio/blob/dff2dacde9a5eb84e439b9aa10ddeb4d4667a4e8/pkg/config/mesh/mesh_test.go#L87?

Reply:
Sorry you're right. Missed it.
And yet   ApplyMeshConfig comment ->`ApplyMeshConfig returns a new MeshConfig decoded from the
 input YAML with the provided defaults applied to omitted configuration values.`
 ApplyMeshConfig suppose to return new instance, not to edit-in-place. that's why there was a check in test to '// Ensure we didn't modify the passed in mesh config'

Reply:
ApplyMeshConfig should base it's output upon the second param, `defaultConfig`, no to modify it!